lib/src/router/mod.rs

```diff
@@ -419,7 +420,13 @@ impl DomainRule {
                     && !&hostname[..len_without_suffix].contains(&b'.')
             }
             DomainRule::Exact(s) => s.as_bytes() == hostname,
-            DomainRule::Regex(r) => r.is_match(hostname),
+            DomainRule::Regex(r) => {
+                let start = Instant::now();
+                let res = r.is_match(hostname);
+                let now = Instant::now();
+                time!("regex_matching_time", (now - start).whole_milliseconds());
+                res
```

Comment:
`res` is mileading. Please rename to `is_a_match` or anything that conveys the idea of a boolean.